extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -213,8 +213,10 @@ void register(
             if (value != null) {
                 // Add the type-id-resolver class
                 reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, value.asClass().name().toString()));
-                // Add the whole hierarchy of the annotated class
-                addReflectiveHierarchyClass(resolverInstance.target().asClass().name(), reflectiveHierarchyClass);
+                if (resolverInstance.target().kind() == CLASS) {
```

Comment:
I wonder if we should also handle the other cases here? I.e. register the targets for other kinds? Not exactly sure though.

Reply:
The problem is that I am not exactly sure how they should be handled...

Reply:
I can understand that. Let's leave it at that then.